tests/nfv/trex_runner.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Trex traffic generator runner
+#
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "opensusebasetest";
+use testapi;
+use strict;
+use utils;
+use lockapi;
+use mmapi;
+
+sub run {
+    my $trex_dir = "/tmp/trex-core";
+
+    select_console 'root-ssh';
+
+    record_info("Waiting for VSPerf installation..");
+    mutex_wait('NFV_VSPERF_READY');
+
+    type_string("nohup sh /root/script.sh &\n");
+    record_info("Started T-REX traffic generator. Waiting for VSPerf to finish tests...");
+
+    mutex_wait('NFV_TESTING_DONE');
+
+    record_info("Upload VSPerf logs");
+    assert_script_run("cd /tmp");
+    assert_script_run(q(find . -type d -name "results*"|xargs -d "\n" tar -czvf vsperf_logs.tar.gz));
+    upload_logs('vsperf_logs.tar.gz', failok => 1);
+}
+
+sub test_flags {
+    return {fatal => 1};
```

Comment:
does it have to be fatal?

Reply:
I'd say so. If this module doesn't work, the test can't be performed at all.